usaspending_api/search/delta_models/award_search.py

```diff
@@ -403,35 +391,41 @@
     GROUP BY
         GROUPED_BY_DEFC.award_id
 ) DEFC on DEFC.award_id = awards.id
-LEFT JOIN (
+LEFT OUTER JOIN (
   SELECT
     faba.award_id,
     COLLECT_SET(
       DISTINCT CONCAT(
-        'agency=', agency.toptier_code,
-        'faaid=', fa.agency_identifier,
-        'famain=', fa.main_account_code,
-        'aid=', taa.agency_id,
-        'main=', taa.main_account_code,
-        'ata=', taa.allocation_transfer_agency_id,
-        'sub=', taa.sub_account_code,
-        'bpoa=', taa.beginning_period_of_availability,
-        'epoa=', taa.ending_period_of_availability,
-        'a=', taa.availability_type_code
+        'agency=', COALESCE(agency.toptier_code, ''),
+        'faaid=', COALESCE(fa.agency_identifier, ''),
+        'famain=', COALESCE(fa.main_account_code, ''),
+        'aid=', COALESCE(taa.agency_id, ''),
+        'main=', COALESCE(taa.main_account_code, ''),
+        'ata=', COALESCE(taa.allocation_transfer_agency_id, ''),
+        'sub=', COALESCE(taa.sub_account_code, ''),
+        'bpoa=', COALESCE(taa.beginning_period_of_availability, ''),
+        'epoa=', COALESCE(taa.ending_period_of_availability, ''),
+        'a=', COALESCE(taa.availability_type_code, '')
       )
-    ) tas_paths,
+    ) AS tas_paths,
     COLLECT_SET(
-      DISTINCT CONCAT(
-        'aid=', taa.agency_id,
-        'main=', taa.main_account_code,
-        'ata=', taa.allocation_transfer_agency_id,
-        'sub=', taa.sub_account_code,
-        'bpoa=', taa.beginning_period_of_availability,
-        'epoa=', taa.ending_period_of_availability,
-        'a=', taa.availability_type_code
+      CONCAT(
+        'aid=', COALESCE(taa.agency_id, ''),
+        'main=', COALESCE(taa.main_account_code, ''),
+        'ata=', COALESCE(taa.allocation_transfer_agency_id, ''),
+        'sub=', COALESCE(taa.sub_account_code, ''),
+        'bpoa=', COALESCE(taa.beginning_period_of_availability, ''),
+        'epoa=', COALESCE(taa.ending_period_of_availability, ''),
+        'a=', COALESCE(taa.availability_type_code, '')
       )
```

Comment:
Since this is what the data currently is in `transaction_search` I left it as is. Should there be a separator? Maybe. Do I want to add a separator and spend the time making sure it doesn't break anything? No.